sdk/lib/cmlib/hiveinit.c

```diff
@@ -44,6 +45,129 @@ HvpVerifyHiveHeader(
     return TRUE;
 }
 
+BOOLEAN CMAPI
+HvRestoreFromLog(
+    _In_ PHHIVE Hive,
+    _Inout_ PHBASE_BLOCK BaseBlock
+)
+{
+    HBASE_BLOCK logBaseBlock;
+    ULONG offset = 0;
+    ULONG indexInLog;
+    BOOLEAN isSuccess;
+    UCHAR dirtyVectBuffer[HSECTOR_SIZE];
+    UCHAR buffer[HBLOCK_SIZE];
+
+    ASSERT(sizeof(HBASE_BLOCK) >= (HSECTOR_SIZE * Hive->Cluster));
+
+    // Read log file header
+    isSuccess = Hive->FileRead(Hive,
+                            HFILE_TYPE_LOG,
+                            &offset,
+                            &logBaseBlock,
```

Comment:
Normally, code will allocate a block, like there:
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/cmlib/hiveinit.c;hb=a8d2cd4b9d124bf9a9d97105cb11923827d7142b#l386
to be used in this `FileRead` call afterwards.